src/parsers/manifestjson.js

```diff
@@ -234,6 +236,28 @@ export default class ManifestJSONParser extends JSONParser {
     }
   }
 
+  async validateIcon(iconPath, expectedSize) {
+    try {
+      const info = await getImageMetadata(this.io, iconPath);
+      if (info.width !== info.height) {
+        this.collector.addError(messages.iconIsNotSquare(iconPath));
+        this.isValid = false;
+      } else if (info.mime !== 'image/svg+xml' &&
+                  parseInt(info.width, 10) !== parseInt(expectedSize, 10)) {
+        this.collector.addWarning(messages.iconSizeInvalid({
+          path: iconPath,
+          expected: parseInt(expectedSize, 10),
+          actual: parseInt(info.width, 10),
+        }));
+      }
+    } catch (err) {
+      this.collector.addWarning(messages.corruptIconFile({
+        path: iconPath,
+      }));
+      log.debug('Icon validation message', err);
```

Comment:
Nit, do you mind to move the `log.debug` call before the `this.collector.addWarning` call?

also, how about changing the message from 'Icon validation message' into something like `Unexpected error raised while validating icon "${iconPath}"`? 

(so that it makes it clear that we are logging the underlying error and not the related validation message, and we also ensure that the `iconPath` is part of the logged message, there are chances that the `iconPath` could be already part of the raised error message but we can't be 100% sure because it is an unexpected exception).

Reply:
@rpl it's done. Please take a look.